packages/headless/src/controllers/core/context/headless-core-internal-context.ts

```diff
@@ -44,10 +49,16 @@ export function buildInternalCoreContext(engine: CoreEngine): InternalContext {
   }
   const {dispatch} = engine;
   return {
-    add(contextKey: string, contextValue: ContextValue) {
+    add(contextKey: ReservedContextKey, contextValue: ContextValue) {
+      if (!isReservedContextKey(contextKey)) {
+        throw new UnreservedContextKeyError(contextKey);
+      }
       dispatch(addContext({contextKey, contextValue}));
```

Comment:
I dont understand how the throw would happen here if this is internal? How does someone 'reserve' a key ? This would not be client facing right ?

Reply:
Contributors to the repo (like our frend from @coveo/salesforce-integration) will reserve a key by modifying the
`ReservedContextKeys` array.
Customer can't reserve keys. They have to respect them tho, otherwise 💥 in their page 😈